Assets/SEE/XR/RadialSelection.cs

```diff
@@ -0,0 +1,409 @@
+            // preorder traversal.
+            if (parent != null)
+            {
+                if (parent is ActionStateTypeGroup parentGroup)
+                {
+                    var search = subMenus.FirstOrDefault(item => item.Item1 == parent.Name);
+                    for (int i = 0; i < subMenus.Count; i++)
+                    {
+                        if (subMenus[i].Item1 == parent.Name && subMenus[i].Item2 == null)
+                        {
+                            subMenus[i] = (subMenus[i].Item1, new());
+                            subMenus[i].Item2.Add(name);
+                        }
+                        else if (subMenus[i].Item1 == parent.Name && subMenus[i].Item2 != null)
+                        {
+                            subMenus[i].Item2.Add(name);
+                        }
+                    }
+                }
+                else
+                {
+                    throw new System.InvalidCastException($"Parent is expected to be an {nameof(ActionStateTypeGroup)}.");
+                }
+            }
+            else
+            {
+                subMenus.Add((name, null));
+                actions.Add(name);
+                numberOfRadialPart += 1;
+            }
+            // Continue with the traversal.
+            return true;
+        }
+    }
+    /// <summary>
+    /// Is true, when the radial menu is open.
+    /// </summary>
+    bool radialMenuTrigger;
+    /// <summary>
+    /// This method gets called, when the button for the radial-menu is pressed.
+    /// </summary>
+    /// <param name="context">Information provided to action callbacks about what triggered an action.</param>
+    private void RadialMenu(InputAction.CallbackContext context)
+    {
+        radialMenuTrigger = true;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (radialMenuTrigger && !spawned)
+        {
+            SpawnRadialPart();
+            radialMenuTrigger = false;
+        }
+        if (spawned)
+        {
+            GetSelectedRadialPart();
+        }
+        if (radialMenuTrigger && spawned)
+        {
+            spawned = false;
+            radialMenuTrigger = false;
+            HideAndTriggerSelected();
+        }
+    }
+    /// <summary>
+    /// Is true, when the current action changed.
+    /// It is being used, to trigger the update for the GlobalActionHistory.
+    /// </summary>
+    public static bool IndicatorChange { set; get; }
+    /// <summary>
+    /// This method activates the selected action and deactivates the radial menu.
+    /// </summary>
+    public void HideAndTriggerSelected()
+    {
+        radialPartCanvas.gameObject.SetActive(false);
+        OnPartSelected.Invoke(currentSelectedRadialPart);
+        IndicatorChange = true;
+    }
+    /// <summary>
+    /// This method calculates, at which radial-part the user is aiming at.
+    /// </summary>
+    public void GetSelectedRadialPart()
```

Comment:
The documentation and especially the name seems to indicate that the method returns some value. Please update the documentation to explain more clearly what happens (i.e., that the value is saved in `currentSelectedRadialPart`) and change the name to something more fitting (like `UpdateSelectedRadialPart`).